compiler/src/iree/compiler/Codegen/Common/ConvertUnsupportedFloatArithToF32.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright 2025 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//===- BubbleUpOrdinalOpPass.cpp -----------------------------------------===//
+//
+// The workgroup count computation when using slices needs the ordinal
+// annotation ops to be bubbled up as much as possible. This pass implements
+// patterns to bubble these operations up.
+//
+//===---------------------------------------------------------------------===//
+
+#include "iree/compiler/Codegen/Common/Passes.h"
+#include "iree/compiler/Codegen/Common/Transforms.h"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Arith/Transforms/Passes.h"
+#include "mlir/IR/BuiltinTypes.h"
+
+namespace mlir::iree_compiler {
+
+#define GEN_PASS_DEF_CONVERTUNSUPPORTEDFLOATARITHTOF32PASS
+#include "iree/compiler/Codegen/Common/Passes.h.inc"
+
+namespace {
+
+struct ConvertUnsupportedFloatArithToF32Pass final
+    : public impl::ConvertUnsupportedFloatArithToF32PassBase<
+          ConvertUnsupportedFloatArithToF32Pass> {
+  void runOnOperation() override;
+};
+
+} // namespace
+
+void ConvertUnsupportedFloatArithToF32Pass::runOnOperation() {
+
+  MLIRContext *context = &getContext();
+  Operation *op = getOperation();
+
+  // Add the source types to be converted to the target type.
+  SmallVector<Type> sourceTypes = {Float8E4M3FNUZType::get(context)};
```

Comment:
As a thought, could we merge the BF16 -> F32 arithmetic pass into this one?

Also, you'll also want the E5M2 type here